File: scala-dci/src/example/domain/Account.java
package example.domain;

import example.dci.Actor;

public class Account implements Actor {
	
	private double balance = 0;

	public Account(double balance) {
		this.balance = balance;
	}
	
	public double getBalance() {
		return balance;
	}

	public void increaseBalance(double amount) {
		balance += amount;
	}
	
	public void decreaseBalance(double amount) {
		balance -= amount;
	}
}

File: scala-dci/src/example/domain/Account.scala
package example.domain

trait Account {
  
  private var balance: Double = 0
  
  def availableBalance: Double = balance 
  
  def decreaseBalance(amount: Double) {
      if (amount < 0)
        throw new IllegalArgumentException
      balance -= amount
  }
  
  def increaseBalance(amount: Long) { 
    balance += amount
  }
  
}